doc.txt

```diff
@@ -1,3 +1,20 @@
+/**
+ * @defgroup config Compile time configurations
+ * @brief    List of compile-time configurations
+ *
+ *           This group contains all compile-time configurations macros that are
+ *           intended to be modified by the user.
+ *
+ *           These configuration macros are defined within modules like this:
+ *           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ [.c] 
+ *           #ifndef SOME_CONFIGURATION_PARAM
+ *           #define SOME_CONFIGURATION_PARAM DEFAULT_VALUE
+ *           #endif
+ *           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *           Configurations can be altered e.g via CFLAGS (e.g
```

Comment:
Maybe add something like "Because of that, configuration can be altered […]". Otherwise, people not that familiar with C might not know, what the explanation above is good for.

Reply:
yes, you are right. I will